outputs.tf

```diff
@@ -161,3 +161,102 @@ output "helm_releases" {
   description = "Map of attributes of the Helm release created"
   value       = helm_release.this
 }
+
+################################################################################
+# GitOps Bridge
+################################################################################
+
+output "gitops_metadata" {
```

Comment:
Two parts on this:
1. Lets not pass through - only export what is created here
2. Don't make users check for values later; if the resource isn't created, then don't export anything about it. Take a look at below:

```hcl
output "gitops_metadata" {
  description = "GitOps Bridge metadata"
  value = merge(
    { for k, v in {
      iamRoleArn     = module.aws_cloudwatch_metrics.iam_role_arn
      namespace      = local.aws_cloudwatch_metrics_namespace
      serviceAccount = local.aws_cloudwatch_metrics_service_account
      } : "awsCloudwatchMetrics_${k}" => v if var.enable_aws_cloudwatch_metrics
    },
    { for k, v in {
      iamRoleArn               = module.aws_efs_csi_driver.iam_role_arn
      namespace                = local.aws_efs_csi_driver_namespace
      controllerServiceAccount = local.aws_efs_csi_driver_controller_service_account
      nodeServiceAccount       = local.aws_efs_csi_driver_node_service_account
      } : "awsEfsCsiDriver_${k}" => v if var.enable_aws_efs_csi_driver
    },
    { for k, v in {
      iamRoleArn     = module.aws_for_fluentbit.iam_role_arn
      namespace      = local.aws_for_fluentbit_namespace
      serviceAccount = local.aws_for_fluentbit_service_account
      logGroupName   = try(aws_cloudwatch_log_group.aws_for_fluentbit[0].name, "")
      } : "awsForFluentbit_${k}" => v if var.enable_aws_for_fluentbit
    },
    { for k, v in {
      iamRoleArn               = module.aws_fsx_csi_driver.iam_role_arn
      namespace                = local.aws_fsx_csi_driver_namespace
      controllerServiceAccount = local.aws_fsx_csi_driver_controller_service_account
      nodeServiceAccount       = local.aws_fsx_csi_driver_node_service_account
      } : "awsFsxCsiDriver_${k}" => v if var.enable_aws_fsx_csi_driver
    },
    { for k, v in {
      iamRoleArn     = module.aws_gateway_api_controller.iam_role_arn
      namespace      = local.aws_gateway_api_controller_namespace
      serviceAccount = local.aws_gateway_api_controller_service_account
      } : "awsGatewayApiController_${k}" => v if var.enable_aws_gateway_api_controller
    },
    { for k, v in {
      iamRoleArn     = module.aws_load_balancer_controller.iam_role_arn
      namespace      = local.aws_load_balancer_controller_namespace
      serviceAccount = local.aws_load_balancer_controller_service_account
      } : "awsLoadBalancerController_${k}" => v if var.enable_aws_load_balancer_controller
    },
    { for k, v in {
      iamRoleArn     = module.aws_node_termination_handler.iam_role_arn
      namespace      = local.aws_node_termination_handler_namespace
      serviceAccount = local.aws_node_termination_handler_service_account
      sqsQueueUrl    = module.aws_node_termination_handler_sqs.queue_url
      } : "awsNodeTerminationHandler_${k}" => v if var.enable_aws_node_termination_handler
    },
    { for k, v in {
      iamRoleArn     = module.aws_privateca_issuer.iam_role_arn
      namespace      = local.aws_privateca_issuer_namespace
      serviceAccount = local.aws_privateca_issuer_service_account
      } : "awsPrivateCaIssuer_${k}" => v if var.enable_aws_privateca_issuer
    },
    { for k, v in {
      iamRoleArn     = module.cert_manager.iam_role_arn
      namespace      = local.cert_manager_namespace
      serviceAccount = local.cert_manager_service_account
      } : "certManager_${k}" => v if var.enable_cert_manager
    },
    { for k, v in {
      iamRoleArn     = module.cluster_autoscaler.iam_role_arn
      namespace      = local.cluster_autoscaler_namespace
      serviceAccount = local.cluster_autoscaler_service_account
      } : "clusterAutoscaler_${k}" => v if var.enable_cluster_autoscaler
    },
    { for k, v in {
      iamRoleArn     = module.external_dns.iam_role_arn
      namespace      = local.external_dns_namespace
      serviceAccount = local.external_dns_service_account
      } : "externalDns_${k}" => v if var.enable_external_dns
    },
    { for k, v in {
      iamRoleArn     = module.external_secrets.iam_role_arn
      namespace      = local.external_secrets_namespace
      serviceAccount = local.external_secrets_service_account
      } : "externalSecrets_${k}" => v if var.enable_external_secrets
    },
    { for k, v in {
      iamRoleArn              = module.karpenter.iam_role_arn
      namespace               = local.karpenter_namespace
      serviceAccount          = local.karpenter_service_account_name
      sqsQueueName            = module.karpenter_sqs.queue_name
      nodeInstanceProfileName = local.karpenter_node_instance_profile_name
      } : "karpenter_${k}" => v if var.enable_karpenter
    },
    { for k, v in {
      iam_role_arn    = module.velero.iam_role_arn
      namespace       = local.velero_namespace
      service_account = local.velero_service_account
      } : "velero_${k}" => v if var.enable_velero
    }
  )
}
```

Reply:
and I think below is the form that should work for https://github.com/gitops-bridge-dev/gitops-bridge/blob/main/argocd/iac/terraform/modules/gitops-bridge-metadata/main.tf (I had a local copy when I was playing around, but didn't plug back into the actual thing to test)

```hcl
locals {
  labels = {
    cluster_name  = var.cluster_name,
    environment   = var.environment
    enable_argocd = true
  }

  annotations = merge(
    local.labels,
    var.metadata,
    var.addons
  )

  argocd = {
    apiVersion = "v1"
    kind       = "Secret"
    metadata = {
      name        = try(var.argocd.secret_name, var.cluster_name)
      namespace   = try(var.argocd.secret_namespace, "argocd")
      annotations = { for key, value in local.annotations : key => tostring(value) if value != null }
      labels = merge(
        local.labels,
        { "argocd.argoproj.io/secret-type" = "cluster" }
      )

    }
    stringData = {
      name   = var.cluster_name
      server = try(var.argocd.server, "https://kubernetes.default.svc")
      config = try(var.argocd.argocd_server_config, {
        tlsClientConfig = {
          insecure = false
        }
      })
    }
  }

  fluxcd = {
    apiVersion = "v1"
    kind       = "ConfigMap"
    metadata = {
      name      = try(var.fluxcd.configmap_name, var.cluster_name)
      namespace = try(var.fluxcd.configmap_namespace, "flux-system")
    }
    data = { for key, value in local.cluster_config : key => tostring(value) if value != null }
  }
}
```

Reply:
Thanks for the review I will look into it and refactor, I would keep the snake case for the output keys

Reply:
@bryantbiggs I refactored the output, the metadata module, and all the examples, please take a look when you have a chance